surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/RunListenerAdapter.java

```diff
@@ -124,13 +125,16 @@ public void executionFinished( TestIdentifier testIdentifier, TestExecutionResul
                     }
                     break;
                 case FAILED:
+                    String reason = safeGetMessage( testExecutionResult.getThrowable().orElse( null ) );
                     if ( isAssertionError )
                     {
-                        runListener.testFailed( createReportEntry( testIdentifier, testExecutionResult, elapsed ) );
+                        runListener.testFailed( createReportEntry( testIdentifier, testExecutionResult, 
+                                reason, elapsed ) );
                     }
                     else
                     {
-                        runListener.testError( createReportEntry( testIdentifier, testExecutionResult, elapsed ) );
+                        runListener.testError( createReportEntry( testIdentifier, testExecutionResult, 
```

Comment:
The method `createReportEntry( testIdentifier, testExecutionResult, reason, elapsed )` is called twice.
Maybe you can call it once and move it between the Line 128 and 129.

Reply:
done